common/src/main/java/com/wynntils/gui/render/FontRenderer.java

```diff
@@ -82,6 +87,67 @@ public void renderText(
         }
     }
 
+    private void renderRainbowText(
+            PoseStack poseStack,
+            String text,
+            float x,
+            float y,
+            HorizontalAlignment horizontalAlignment,
+            VerticalAlignment verticalAlignment,
+            TextShadow shadow) {
+        float renderX;
+        float renderY;
+
+        renderX = switch (horizontalAlignment) {
+            case Left -> x;
+            case Center -> x - (font.width(text) / 2f);
+            case Right -> x - font.width(text);};
+
+        renderY = switch (verticalAlignment) {
+            case Top -> y;
+            case Middle -> y - (font.lineHeight / 2f);
+            case Bottom -> y - font.lineHeight;};
+
+        for (char c : text.toCharArray()) {
+            long dif = ((long) renderX * 10) - ((long) renderX * 10);
+
+            String s = String.valueOf(c);
```

Comment:
This will break with chatformatting codes. You need to strip color codes beforehand, but you have to leave italic bold reset etc.

Reply:
Reminder (this is undocumented lol): ChatFormatting color codes always override the color you pass to font renderer.